File: wind-recruitment-api/src/main/java/com/wind/windrecruitmentapi/entities/Topic.java
package com.wind.windrecruitmentapi.entities;


import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Topic {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    private String name;
    private String description;
    private String duration;

    @ElementCollection
    private List<String> requirements;

    @ManyToOne
    @JoinColumn(
            name = "hr_recruiter_id",
            nullable = false,
            referencedColumnName = "id"
    )
    private HRRecruiter topicAuthor;

}

File: wind-recruitment-ui/src/app/services/models/topic.ts
/* tslint:disable */
/* eslint-disable */
import { HrRecruiter } from '../models/hr-recruiter';
export interface Topic {
  description?: string;
  duration?: number;
  id?: number;
  name?: string;
  requirements?: Array<string>;
  topicAuthor?: HrRecruiter;
}
